### PR TITLE
Do not allow changing release setting when item has been released.

### LINKED ISSUE #1741: When versioning an item, only display release options when the item has an embargo date that has not yet passed.
See previous ticket #1352

I have tested this again and it is still not working. Behavior is slightly different this time. Changes are in bold below.

Yesterday I created https://sul-h2-stage.stanford.edu/works/18075
 and set it for embargo release today.

- Today I checked that embargo was lifted. H2 had a history entry saying it was released, but **the PURL still has the embargo message and will not allow me to download the file.**
- I then attempted to version the item. I opened it, added a version comment, added two files, and added in new keywords (removing the one old one).
- When I clicked deposit it behaved it first triggered an **error notification that the embargo date was invalid**. This should not happen for an item released from embargo. 
- Once I changed the embargo date and clicked "submit for review", it reloaded the page with the version comments gone and no error messages. My changes had also been wiped out.

This also triggered the HB alert pasted in below.

Error detected in: happy-heron (stage)

Full report at https://app.honeybadger.io/projects/77112/faults/80069649

NoMethodError: undefined method `[]=' for nil:NilClass

Summary

works#update

URL: https://sul-h2-stage.stanford.edu/works/18075

Hostname: sul-h2-stage.stanford.edu

Context

user_scope: user
user_id: 12
user_email: amyhodge@stanford.edu
Trace

app/services/new_version_parameter_filter.rb:19:in `block in filter_attached_files`
app/services/new_version_parameter_filter.rb:17:in `filter_attached_files`
app/services/new_version_parameter_filter.rb:7:in `call`

### PR DESCRIPTION
**Pull request by justinlittman**
closes #1741

## Why was this change made?
Per ticket.

## How was this change tested?
Unit

## Which documentation and/or configurations were updated?
NA

### CODE REVIEW

**Review by ndushay — APPROVED** [review]
approving but wondering if Amy needs to test

**Comment by justinlittman** [comment]
Amy typically tests after merge and on stage.

**Pull request merged**
